### ISSUE TITLE
cxxrtl blackbox behavior issue / possible bug ?

### ISSUE THREAD

**Issue by smunaut** [YosysHQ/yosys#2887]
Attached is an example showing the issue.  (sorry about tar.bz2 inside zip ... github doesn't want tar.bz2 attach :/ )

[cxxrtl_bb.zip](https://github.com/YosysHQ/yosys/files/6872767/cxxrtl_bb.zip)

It's a simple memory block implemented two ways : once using pure verilog  ( `mem_sim_data_verilog.v` ) and the other as a `cxxrtl` blacbox ( `blackbox.cpp` & `mem_sim_data_bb.v` ) and I expect them both to produce the same behavior, but they don't.

I didn't even get to test the write port but the read port appears as a "async" read instead of a "sync" one (i.e. data appears directly and not at the next cycle like I'd expect).

**Comment by whitequark** [comment]
This is a bug; there is currently no way to make the blackbox version work.

**Comment by smunaut** [comment]
I'm wondering : was that ever solved ? I know at the time there was talk about some large rewrite/reorganization that would take care of it, but I didn't follow if that ever happened ?

**Comment by whitequark** [comment]
No, I left the job that would've paid for it. I will do it when porting cxxrtl to prjunnamed.
